packages/web-components/src/dialog-body/dialog-body.template.ts

```diff
@@ -0,0 +1,50 @@
+import { elements, ElementViewTemplate, html, ref, slotted, when } from '@microsoft/fast-element';
+import { DialogType } from '../dialog/dialog.options.js';
+
+const dismissed16Regular = html.partial(`
+  <svg
+    fill="currentColor"
+    aria-hidden="true"
+    width="20"
+    height="20"
+    viewBox="0 0 20 20"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <path
+      d="m4.09 4.22.06-.07a.5.5 0 0 1 .63-.06l.07.06L10 9.29l5.15-5.14a.5.5 0 0 1 .63-.06l.07.06c.18.17.2.44.06.63l-.06.07L10.71 10l5.14 5.15c.18.17.2.44.06.63l-.06.07a.5.5 0 0 1-.63.06l-.07-.06L10 10.71l-5.15 5.14a.5.5 0 0 1-.63.06l-.07-.06a.5.5 0 0 1-.06-.63l.06-.07L9.29 10 4.15 4.85a.5.5 0 0 1-.06-.63l.06-.07-.06.07Z"
+      fill="currentColor"
+    ></path>
+  </svg>`);
+
+/**
+ * Template for the dialog form
+ * @public
+ */
+export const template: ElementViewTemplate = html`
+  <div class="title" part="title">
+    <!--
+--><slot name="title"></slot
+    ><!--
+--><slot ${slotted({ property: 'titleAction', filter: elements() })} name="title-action"></slot
+    ><!--
+-->${when(
+      // if parent is is non-modal and no title action is provided and noTitleAction is false, show default action
+      x => x.parentNode?.type === DialogType.nonModal && x.titleAction.length === 0 && !x.noTitleAction,
+      html`<fluent-button
+        tabindex="0"
+        part="title-action"
+        class="title-action"
+        appearance="transparent"
+        icon-only
+        @click=${x => x.parentNode?.hide()}
+        ${ref('defaultTitleAction')}
+      >
+        ${dismissed16Regular}
+      </fluent-button>`,
+    )}
```

Comment:
I'm curious, could we remove this when by having this simply be the default slotted element for `title-action` and hiding it in the cases where it shouldn't exist (I think this can exist in modal instances but not alert...)? Any slotted content would project in its place, we can likely avoid checking for slotted elements in general in JS and move to just CSS either based on the `type` attribute being `alert`. Thoughts? Would that work?

Reply:
I _think_ the historical intent here is that because non-modal dialogs don't have light-dismiss, an author could theoretically create a non-modal dialog that could never be closed. 😱 This provides a default "pit of success" in a way. 

That said... I think you're right and making `title-action` BYO might be the most elegant way forward here, rather than trying to infer intent and creating a bit of prop-soup to manage that. It would simplify our props surface and Storybook stories considerably.

Reply:
I think we could still (if we want to include it) provide a default slotted mechanism here for convenience. The "golden path" would be to use the fluent-button generally. I just think that makes it so we can use CSS based on the role or the `type="foo"` attribute selector to just *hide it* when we shouldn't provide it.

Reply:
@chrisdholt I refactored this a bit and it's a lot cleaner with the default slotted content. I couldn't get CSS-only approach to work because `:host` doesn't know what `type` the parent `<fluent-dialog>` is so instead I'm setting a `[hidden]` attribute based on the instance and it's not zero-JS but it's a lot simpler than before. Let me know if you want me to push on this further.

Reply:
> @chrisdholt I refactored this a bit and it's a lot cleaner with the default slotted content. I couldn't get CSS-only approach to work because `:host` doesn't know what `type` the parent `<fluent-dialog>` is so instead I'm setting a `[hidden]` attribute based on the instance and it's not zero-JS but it's a lot simpler than before. Let me know if you want me to push on this further.

A great use for `:host-context()` which has near zero support outside of Chrome 😆